src/parsers/installrdf.js

```diff
@@ -1,66 +1,77 @@
-import { ADDON_TYPE_MAP, INSTALL_RDF, RDF_DEFAULT_NAMESPACE } from 'const';
-import { TYPE_INVALID, TYPE_MISSING } from 'messages';
+import { ADDON_TYPE_MAP, RDF_DEFAULT_NAMESPACE } from 'const';
+import { RDF_TYPE_INVALID, RDF_NAME_MISSING,
+         RDF_TYPE_MISSING } from 'messages';
 import log from 'logger';
-import RDFScanner from 'scanners/rdf';
 
 
 export default class InstallRdfParser {
 
-  constructor(rdfString, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
-    this.rdfString = rdfString;
+  constructor(xmlDoc, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
+    this.xmlDoc = xmlDoc;
     // Provides ability to directly add messages to
     // the collector.
     this.collector = collector;
     this.namespace = namespace;
   }
 
-  parseDoc() {
-    var rdfScanner = new RDFScanner(this.rdfString, INSTALL_RDF);
-    return rdfScanner.getContents();
-  }
-
   getMetaData() {
-    return this.parseDoc()
-      .then((xmlDoc) => {
-        return Promise.resolve({
-          guid: this._getGUID(xmlDoc),
-          type: this._getAddonType(xmlDoc),
-        });
-      });
+    return Promise.resolve({
+      guid: this._getGUID(),
+      name: this._getName(),
+      type: this._getAddonType(),
+    });
   }
 
-  _getAddonType(xmlDoc) {
-    var addonType = null;
-    var typeNodes = xmlDoc.getElementsByTagName('em:type');
+  _getNode(tag) {
+    var typeNodes = this.xmlDoc.getElementsByTagName(tag);
     if (typeNodes.length > 1) {
-      throw new Error('Multiple <em:type> elements found');
+      throw new Error(`Multiple <${tag}> elements found`);
     }
-    var node = typeNodes[0];
+    return typeNodes[0];
+  }
+
+  _getAddonType() {
+    var addonType = null;
+    var node = this._getNode('em:type');
     if (node && node.firstChild && node.firstChild.nodeValue) {
       var typeValue = node.firstChild.nodeValue;
       if (!ADDON_TYPE_MAP.hasOwnProperty(typeValue)) {
         log.debug('Invalid type value "%s"', typeValue);
-        this.collector.addError(TYPE_INVALID);
+        this.collector.addError(RDF_TYPE_INVALID);
       } else {
-        var addonType = ADDON_TYPE_MAP[typeValue];
+        addonType = ADDON_TYPE_MAP[typeValue];
         log.debug('Mapping original <em:type> value "%s" -> "%s"',
                   typeValue, addonType);
       }
     } else {
       log.warn('<em:type> was not found in install.rdf');
-      this.collector.addNotice(TYPE_MISSING);
+      this.collector.addNotice(RDF_TYPE_MISSING);
     }
     return addonType;
   }
 
-  _getGUID(xmlDoc) {
-    if (xmlDoc.getElementsByTagNameNS(this.namespace, 'id').length > 0) {
-      var idNode = xmlDoc.getElementsByTagNameNS(this.namespace, 'id').item(0);
+  _getGUID() {
+    // Note: this has shared validiation in the `Validator.getAddonMetaData`
```

Comment:
This caught me off-guard at first. I think the comment needs a bit more clarity:

```js
// NOTE: We validate this rule in one place: `Validator.getAddonMetaData`.
// This is because both `install.rdf` and `manifest.json` share the same requirements
// for guid.
```

Wordier, but clearer?